engine/render/renderer/platform/vulkan/CommandBuffer.h

```diff
@@ -0,0 +1,151 @@
+//
+//  Copyright (c) 2022 Jonathan Moallem (@J-Mo63) & Aryeh Zinn (@Raelr)
+//
+//  This code is released under an unmodified zlib license.
+//  For conditions of distribution and use, please see:
+//      https://opensource.org/licenses/Zlib
+//
+
+#ifndef SIEGE_ENGINE_VULKAN_COMMANDBUFFER_H
+#define SIEGE_ENGINE_VULKAN_COMMANDBUFFER_H
+
+#include <utils/collections/HeapArray.h>
+
+#include "utils/Types.h"
+
+namespace Siege::Vulkan
+{
+/**
+ * The CommandBuffer Wrapper class. This class represents either a single or a collection of
+ * CommandBuffer objects. CommandBuffers allow us to record commands for each frame of execution
+ */
+class CommandBuffer
+{
+public:
+
+    /**
+     * Default constructor for the CommandBuffer. Initialises all values as empty
+     */
+    CommandBuffer() = default;
+
+    /**
+     *  Default destructor. Destroys all stored vulkan objects by default. Since CommandBuffers are
+     *  destroyed as soon as the CommandPool is destroyed (CommandPool is stored on the
+     * LogicalDevice)
+     */
+    ~CommandBuffer() = default;
+
+    /**
+     * Constructor for the CommandBuffer - Requires a count of the number of buffers to create
+     * @param count the number of buffers to create
+     */
+    CommandBuffer(uint32_t count);
+    /**
+     * Move constructor for the CommandBuffer
+     * @param other the CommandBuffer to move
+     */
+    inline CommandBuffer(CommandBuffer&& other)
+    {
+        Swap(other);
+    }
+
+    /**
+     * Move assignment operator for CommandBuffers
+     * @param other the CommandBuffer to move
+     * @return A reference to the current CommandBuffer
+     */
+    inline CommandBuffer& operator=(CommandBuffer&& other)
+    {
+        Swap(other);
+        return *this;
+    }
+
+    /**
+     * Subscript operator into the stored command buffers
+     * @param idx the index of the CommandBuffer to be accessed
+     * @return the raw Vulkan Command Buffer
+     */
+    inline VkCommandBuffer& operator[](const size_t idx)
+    {
+        return commandBuffers[idx];
+    }
+
+    /**
+     * Returns the raw Vulkan CommandBuffer pointer
+     * @param index the index of the CommandBuffer to be accessed
+     * @return
+     */
```

Comment:
incomplete description, should also contain warning about out of bounds access

Reply:
Adjusted.